bip-tx-terminology.mediawiki

```diff
@@ -0,0 +1,286 @@
+
+; Control Block
+: [TC] The proof of existence for a Leaf Script.  Appears as a Witness Item in Witness Stacks for P2TR script path inputs.  Only consists of the Inner Key for Script Trees with a single Leaf Script, otherwise provides the Inner Key as well as a Taproot Branch in the form of the hashing partners for the used Leaf Script.
+
+; Flag
+: [SA] Serialization artifact indicating features used by the transaction.  As of writing, the only allowed value is <code>01</code> which indicates that the transaction serialization has a Witness Section (see BIP144).
+
+; Forwarding Script
+: [AT] A collective term for scripts that redirect input validation to another script or data structure.  Witness Programs and P2SH Programs are forwarding scripts.
+
+; Input Counter
+: [SA] The length of the transaction input list.
+
+; Input Index
+: [AT] The position of an input in a transaction's Input List.
+
+; Input Script
+: [TC] originally called `scriptSig`. The Bitcoin code base used Hungarian notation when it was published.  This transaction field appearing in Transaction Inputs was named for containing the signature and being of the type script.  Today, this field may be better understood as the _input script_ which is used to satisfy spending conditions for non-segwit outputs, redirects validation to the Witness Stack for wrapped segwit outputs, and is empty for native segwit outputs. The Input Script may contain an actual script when spending a Bare Output Script, but may only contain push operations for any output type since P2SH.
+
+; Leaf Script
+: [TC] A Condition Script used for spending P2TR outputs via the script path.
+
+; Leaf Version
+: [TC] The version of Script used in a leaf script. At time of writing, only _Tapscript_ is defined.
+
+; Locktime
+: [TC] Part of the Transaction Header. The locktime specifies a time (in blockheight or UNIX time) before which a transaction cannot be included in a block if any input's sequence number indicates non-finality.  The locktime can also indicate relative spending lock in combination with `OP_CHECKSEQUENCEVERIFY`. Serialized as the four byte <code>nLockTime</code> integer field.
+
+; Marker
+: [SA] 1-byte serialization artifact indicating that a type of extended serialization is being used for this transaction.  Must always be <code>00</code> (see BIP144).  (Note: Since the serialization of a non-segwit transaction would have the input counter at the marker's position, a whole segwit transaction will always appear invalid to non-segwit nodes since the input counter must be greater than zero.)
```

Comment:
I improved the sentence:

```suggestion
: [SA] 1-byte serialization artifact indicating that a type of extended serialization is being used for this transaction.  Must always be <code>00</code> (see BIP144). (Note: Non-segwit nodes will only accept _stripped segwit transactions_, because the marker appears at the position where non-segwit nodes expect the input counter. The input counter may not be zero, so a complete (non-stripped) segwit transaction appears invalid to a non-segwit node.)

```